packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -1934,22 +1931,14 @@ describe("SharedTree", () => {
 				new TreeViewConfiguration({ schema, enableSchemaValidation: true }),
 			);
 			view.initialize({ foo: ["42"] });
-
-			// Trying to use the tree with a view schema that makes 'foo' an array of strings or numbers
-			// should not cause compile-time errors when inserting a number, but stored schema validation
-			// should kick in and throw an error.
-			const viewschema = sf.object("myObject", {
-				foo: sf.array("foo", [sf.string, sf.number]),
-			});
-			const tree = sharedTree.schematize(
-				new TreeConfiguration(viewschema, () => ({ foo: ["42"] }), {
-					enableSchemaValidation: true,
-				}),
+			assert.throws(
+				() => {
+					// The cast here is necessary as the API provided by `insertAtEnd` is typesafe with respect
+					// to the schema, so in order to insert invalid content we need to bypass the types.
+					view.root.foo.insertAtEnd(3 as unknown as string);
+				},
+				validateUsageError(/Tree does not conform to schema./),
```

Comment:
I talked with @alexvy86 about this test, and we concluded this direction was the most sensible one to move it in.

Note that the 'before' version of this test was a bit misleading: the overload of `assert.throws` which takes in a string as a second argument *does not validate that the error thrown contains that string*, it just uses the string as a message. The intent of the test before was to use `/Tree does not to conform to schema.`, but then the test would have failed as written because accessing `tree.root` fails due to TreeView's compatibility enforcement.